### ISSUE TITLE
Errors when loading an archive

### ISSUE THREAD

**Issue by oliver----** [substance/texture#531]
See the screenshot in https://github.com/stencila/desktop/issues/54#issuecomment-390107842

It seems that there is a problem in how the archive change listener is registered.
What we have:
- Here the archive change listener is registered: https://github.com/substance/texture/blob/45c2e8eb20d4b489507301702eac13f7e6df7332/src/TextureDesktopApp.js#L24
- Here the change listener is implemented: https://github.com/substance/texture/blob/45c2e8eb20d4b489507301702eac13f7e6df7332/src/DesktopAppChrome.js#L42

The reported error from `stencila/desktop` shows that the listener is called before the archive is in the state of the `DesktopAppChrome`

@michael could you please take a look. Maybe it can be solved easily just be swapping the lines in `TextureDesktopApp`, i.e. first load, then register listener

**Comment by michael** [comment]
Thanks for investigating. I'll take care of it! Just finishing off the Reader setup first.

**Comment by michael** [comment]
I'm on this now, do you know how i can reproduce the problem?

**Comment by michael** [comment]
Makes sense. The place where we register the handler is definitely not cool, we need more control. However for now I'd just guard the _archiveChanged handler. Like so:

```js
  _archiveChanged() {
    if (!this.state.archive) return
    const hasPendingChanges = this.state.archive.hasPendingChanges()
    if (hasPendingChanges) {
      this.props.ipc.send('document:unsaved')
      this._updateTitle(hasPendingChanges)
    }
  }
```

that ok?

**Comment by oliver----** [comment]
Yeah, that makes sense, too. ;)

**Comment by michael** [comment]
Will include this in the fix-entity-editing PR. Anything else we want to fix before releasing the patch?

**Comment by oliver----** [comment]
I think we are good now.
